File: app/src/main/java/jp/ttanaka330/androidlearning/ui/fragment/MainFragment.java
package jp.ttanaka330.androidlearning.ui.fragment;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v7.widget.DividerItemDecoration;
import android.support.v7.widget.LinearLayoutManager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.ArrayList;
import java.util.List;

import jp.ttanaka330.androidlearning.databinding.FragmentMainBinding;
import jp.ttanaka330.androidlearning.ui.activity.RealmActivity;
import jp.ttanaka330.androidlearning.ui.activity.RetrofitActivity;
import jp.ttanaka330.androidlearning.ui.view.RecyclerSimpleAdapter;

public class MainFragment extends BaseFragment {

    private FragmentMainBinding mBinding;

    public MainFragment() {
        // Required empty public constructor
    }

    /**
     * {@link MainFragment} のインスタンスを生成します。
     *
     * @return {@link MainFragment} の新規インスタンス
     */
    public static MainFragment newInstance() {
        return new MainFragment();
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        mBinding = FragmentMainBinding.inflate(inflater, container, false);
        initFragmentList();
        return mBinding.getRoot();
    }

    private void initFragmentList() {
        List<Item> list = new ArrayList<>();
        list.add(new Item(RealmActivity.createIntent(getContext())));
        list.add(new Item(RetrofitActivity.createIntent(getContext())));

        RecyclerSimpleAdapter adapter = new RecyclerSimpleAdapter<Item>(list) {
            @Override
            protected void onItemClicked(int position, Item item) {
                startActivity(item.getIntent());
            }
        };

        Context context = getContext();
        LinearLayoutManager layoutManager = new LinearLayoutManager(context);
        DividerItemDecoration decoration = new DividerItemDecoration(context, layoutManager.getOrientation());
        mBinding.listView.setAdapter(adapter);
        mBinding.listView.setLayoutManager(layoutManager);
        mBinding.listView.addItemDecoration(decoration);
    }

    private class Item {
        private final Intent mIntent;

        Item(@NonNull  Intent intent) {
            mIntent = intent;
        }

        @NonNull
        Intent getIntent() {
            return mIntent;
        }

        @Override
        public String toString() {
            assert mIntent.getComponent() != null;
            int pos = mIntent.getComponent().getClassName().lastIndexOf(".") + 1;
            return mIntent.getComponent().getShortClassName().substring(pos);
        }
    }
}

File: app-learning/app/src/main/java/com/github/ttanaka330/android/learning/ui/MainFragment.kt
package com.github.ttanaka330.android.learning.ui

import android.app.Activity
import android.content.Intent
import android.content.pm.PackageManager
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.github.ttanaka330.android.learning.databinding.FragmentMainBinding
import timber.log.Timber

class MainFragment : Fragment() {

    private var _binding: FragmentMainBinding? = null
    private val binding: FragmentMainBinding
        get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentMainBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val adapter = MainRecyclerViewAdapter {
            val intent = Intent().setClassName(view.context, it.path)
            startActivity(intent)
        }
        binding.mainList.adapter = adapter

        activity?.let {
            val item = getLearningInfoList(it)
            adapter.submitList(item)
        }
    }

    private fun getLearningInfoList(parent: Activity): List<LearningInfo>? {
        try {
            val info = parent.packageManager.getPackageInfo(parent.packageName, PackageManager.GET_ACTIVITIES)
            return info.activities?.map {
                val packageName = getPackage(it.name)
                val className = getClassName(it.name)
                LearningInfo(packageName, className)
            }?.filter {
                it.packageName.indexOf(parent.packageName) != -1 &&
                    it.className != parent.javaClass.simpleName
            }
        } catch (e: PackageManager.NameNotFoundException) {
            Timber.e(e)
        }
        return null
    }

    private fun getPackage(fullPath: String): String {
        val lastPos = fullPath.lastIndexOf('.')
        return fullPath.substring(0, lastPos)
    }

    private fun getClassName(fullPath: String): String {
        val lastPos = fullPath.lastIndexOf('.') + 1
        return fullPath.substring(lastPos)
    }
}
